src/main/java/racingcar/dto/RacingCarStatesDTO.java

```diff
@@ -0,0 +1,20 @@
+package racingcar.dto;
+
+import racingcar.RacingCar;
+
+public final class RacingCarStatesDTO {
+    public final RacingCarStateDTO[] catStates;
```

Comment:
오타가 있네요 👀 
```suggestion
    public final RacingCarStateDTO[] carStates;
```

그리고 배열 대신 다양한 기능을 제공하는 List 타입을 활용하면 어떨까요?
이펙티브 자바 아이템 28을 참고해 보시면 좋을 것 같아요.